backend/services/project_search_service.py

```diff
@@ -292,16 +297,13 @@ def _filter_projects(search_dto: ProjectSearchDTO, user):
         query = query.order_by(order_by).group_by(Project.id)
 
         if search_dto.managed_by and user.role != UserRole.ADMIN.value:
-            team_projects = query.join(ProjectTeams).filter(
-                ProjectTeams.role == TeamRoles.PROJECT_MANAGER.value,
-                ProjectTeams.project_id == Project.id,
-            )
-            user_orgs_list = OrganisationService.get_organisations_managed_by_user(
-                search_dto.managed_by
+            # Get all the projects associated with the user and team.
+            user_teams_ids = [u.team_id for u in user.teams]
+            query = (
+                query.filter(ProjectTeams.project_id == Project.id)
+                .filter(ProjectTeams.role == TeamRoles.PROJECT_MANAGER.value)
```

Comment:
This filter overrides the org filter in the `create_search_query` filter. So within `create_search_query` all the proejcts related to user's teams and orgs are collected in one query. Finally within this `managed_by` block when you say filter only those teams that have PM role, all the org related projects are eliminated resulting in null array